### PR TITLE
Bug fix: Models with shared layers shouldn't be considered Sequential like

### PR DESCRIPTION
**Pull request by farizrahman4u**
As of now, the following model is being considered as a "Sequential-like":

```python
from keras.layers import *
from keras.models import *

a = Input((5,))
dense = Dense(5)
b = dense(a)
c = dense(b)

model = Model(a, c)
model.summary()
```

### CODE REVIEW

**Comment by farizrahman4u** [comment]
@fchollet

File: keras/utils/layer_utils.py [hunk 1]
```diff
@@ -24,12 +24,29 @@ def print_summary(model, line_length=None, positions=None, print_fn=print):
...
                 # the model is no longer sequential
                 sequential_like = False
                 break
+            nodes += v
+        if sequential_like:
+            # search for shared layers
+            for layer in model.layers:
+                flag = False
+                for node in layer.inbound_nodes:
+                    if node in nodes:
```
Comment (fchollet) [inline keras/utils/layer_utils.py]: Wouldn't this always be True?

File: keras/utils/layer_utils.py [hunk 2]
```diff
@@ -24,12 +24,29 @@ def print_summary(model, line_length=None, positions=None, print_fn=print):
...
+        nodes = []
+        for v in nodes_by_depth:
             if (len(v) > 1) or (len(v) == 1 and len(v[0].inbound_layers) > 1):
                 # if the model has multiple nodes or if the nodes have multiple inbound_layers
                 # the model is no longer sequential
                 sequential_like = False
                 break
+            nodes += v
+        if sequential_like:
+            # search for shared layers
```
Comment (fchollet) [inline keras/utils/layer_utils.py]: I don't get the criteria you use for shared layers. A comment here would help

File: keras/utils/layer_utils.py [hunk 3]
```diff
@@ -24,12 +24,29 @@ def print_summary(model, line_length=None, positions=None, print_fn=print):
...
+                for node in layer.inbound_nodes:
+                    if node in nodes:
+                        if flag:
+                            sequential_like = False
+                            break
+                        else:
+                            flag = True
+                if not sequential_like:
+                    break
+        del nodes
```
Comment (fchollet) [inline keras/utils/layer_utils.py]: This doesn't seem necessary

File: keras/utils/layer_utils.py (hunk 2 above)
Comment (farizrahman4u) [reply keras/utils/layer_utils.py]: A layer is shared  in a model if it has 2 or more inbound nodes and those inbound nodes are part of the graph.

File: keras/utils/layer_utils.py (hunk 1 above)
Comment (farizrahman4u) [reply keras/utils/layer_utils.py]: Not necessarily. A layer might be shared between 2 models and hence have more than one inbound node. But we are looking for layers which are being shared within the same model.

Comment (farizrahman4u) [reply keras/utils/layer_utils.py]:
```python
dense = Dense(10)
a = Input((10,))
b = dense(a)
model1 = Model(a, b)

a2 = Input((10,))
b2 = dense(a2)
model2 = Model(a2, b2)

model1.summary()  
# Here dense layer has 2 inbound nodes, 
#but the second node is not part of model1.
# So should be considered sequential like

**Pull request merged**